### DISCUSSION TITLE
Storage file upload returning undefined object

### DISCUSSION THREAD

**Discussion by kaustubhuk8** [supabase/supabase#3374] [Questions] [answered]
I am trying to upload a file into a storage bucket using React. The file is uploaded perfectly into the bucket but the data returned to the upload function from supabase is undefined.

const avatarFile = event.target.files[0]
const { data, error } = await supabase.storage
  .from('avatars')
  .upload('public/avatar1.png', avatarFile){
        cacheControl: "3600",
        upsert: false,
      });

in this case, console.log(data) returns undefined. Any thoughts on why this is happening?

**Comment by awalias** [comment, marked as answer]
hey @kaustubhuk8 , which version of `supabase-js` are you using?

it _should_ be returning the path as per https://github.com/supabase/storage-js/blob/f650e1a504bd6137bd94880e37c1675c5eb996b7/src/lib/StorageFileApi.ts#L90 

which is actually just https://github.com/supabase/storage-js/blob/f650e1a504bd6137bd94880e37c1675c5eb996b7/src/lib/StorageFileApi.ts#L322 so you could construct it yourself as a work around if needed

**Reply by kaustubhuk8** [reply]
It is 1.23.0 I believe

**Reply by awalias** [reply]
I see that the `storage-js` version [was bumped in `1.24.0`](https://github.com/supabase/supabase-js/commit/fed12203eb37f61ddb40323c42ad19a8b7abaee2) so maybe try that

**Reply by kaustubhuk8** [reply]
Facing the same issues while trying to download the file using the JS code in the DOCS

**Reply by awalias** [reply]
hmm could you share what polices you have for this bucket, is it publicly readable?

Also feel free to send to support@supabase.com if you don't want to share code here

**Reply by kaustubhuk8** [reply]
Yeah, I think the version change fixed the error.
Sure, I'll send the code to the email.
